modules/component/src/core/operators/coalesce.ts

```diff
@@ -0,0 +1,216 @@
+import {
+  MonoTypeOperatorFunction,
+  Observable,
+  Operator,
+  SubscribableOrPromise,
+  Subscriber,
+  Subscription,
+  TeardownLogic,
+} from 'rxjs';
+import {
+  InnerSubscriber,
+  OuterSubscriber,
+  subscribeToResult,
+} from 'rxjs/internal-compatibility';
+import { generateFrames } from '../projections';
+import { createPropertiesWeakMap } from '../utils/create_properties-weakmap';
+
+export interface CoalesceConfig {
+  context?: object;
+  leading?: boolean;
+  trailing?: boolean;
+}
+
+interface CoalescingContextProps {
+  isCoalescing: boolean;
+}
+
+const coalescingContextPropertiesMap = createPropertiesWeakMap<
+  object,
+  CoalescingContextProps
+>(ctx => ({
+  isCoalescing: false,
+}));
+
+const defaultCoalesceConfig: Pick<CoalesceConfig, 'leading' | 'trailing'> & {
+  context: undefined;
+} = {
```

Comment:
this whole thing is just `CoalesceConfig`.

Reply:
Yes!